orc8r/cloud/go/services/certifier/obsidian/handlers/handlers.go

```diff
@@ -0,0 +1,212 @@
+		{Path: ManageUserTokens, Methods: obsidian.POST, HandlerFunc: addUserTokenHandler},
+		{Path: ManageUserTokens, Methods: obsidian.DELETE, HandlerFunc: deleteUserTokenHandler},
+	}
+	return ret
+}
+
+func listUsersHandler(c echo.Context) error {
+	users, err := certifier.ListUsers(c.Request().Context())
+	if err != nil {
+		return obsidian.MakeHTTPError(err)
+	}
+	return c.JSON(http.StatusOK, users)
+}
+
+func createUserHandler(c echo.Context) error {
+	var data models.User
+	err := json.NewDecoder(c.Request().Body).Decode(&data)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error decoding request body for creating user: %v", err))
+	}
+	username := fmt.Sprintf("%v", *data.Username)
+	password := []byte(fmt.Sprintf("%v", *data.Password))
+	user := &protos.User{
+		Username: username,
+		Password: password,
+	}
+	err = certifier.CreateUser(c.Request().Context(), user)
+	return err
+}
+
+func getUserHandler(c echo.Context) error {
+	username := c.Param("username")
+	user, err := certifier.GetUser(c.Request().Context(), username)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
+	return c.JSON(http.StatusOK, user)
+}
+
+func updateUserHandler(c echo.Context) error {
+	username := c.Param("username")
+	var updatePassword string
+	err := json.NewDecoder(c.Request().Body).Decode(&updatePassword)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error decoding request body for updating user: %v", err))
+	}
+	newUser := &protos.User{Username: username, Password: []byte(updatePassword)}
+	certifier.UpdateUser(c.Request().Context(), newUser)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
+	return nil
+}
+
+func deleteUserHandler(c echo.Context) error {
+	username := c.Param("username")
+	deleteUser := &protos.User{Username: username}
+	err := certifier.DeleteUser(c.Request().Context(), deleteUser)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error deleting user: %v", err))
+	}
+	return nil
+}
+
+func loginHandler(c echo.Context) error {
+	var data models.User
+	err := json.NewDecoder(c.Request().Body).Decode(&data)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error decoding request body for creating user: %v", err))
+	}
+	username := fmt.Sprintf("%v", *data.Username)
+	password := []byte(fmt.Sprintf("%v", *data.Password))
+	user := &protos.User{
+		Username: username,
+		Password: password,
+	}
+	err = certifier.Login(c.Request().Context(), user)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("error logging in: %v", err))
+	}
+	return nil
+}
+
+func getUserTokensHandler(c echo.Context) error {
+	username := c.Param("username")
+	res, err := certifier.ListUserTokens(c.Request().Context(), &protos.User{Username: username})
```

Comment:
Ideally this would return a map of token->policy

Reply:
Yup it does!!!